addons/pos_restaurant/models/pos_config.py

```diff
@@ -42,3 +42,12 @@ def _onchange_is_table_management(self):
     def _onchange_is_order_printer(self):
         if not self.is_order_printer:
             self.printer_ids = [(5, 0, 0)]
+
+    @api.model
+    def get_tables_order_count(self, config_id):
+        """         """
+        config = self.browse(config_id)
+        result = []
+        for table in config.floor_ids.table_ids.search([('active', '=', True)]):
```

Comment:
you have to use config.floor_ids.table_ids.filtered(lambda t: t.active == True) because config.floor_ids.table_ids.search([('active', '=', True)]) will return all table of all config active. It is the same as doing self.env['restaurant.table'].search([])